lib/Elastica/Transport/HttpAdapter.php

```diff
@@ -32,7 +32,7 @@ class HttpAdapter extends AbstractTransport
      *
      * @param Connection $connection
      */
-    public function __construct(Connection $connection = null, HttpAdapterInterface $httpAdapter)
+    public function __construct(?Connection $connection = null, HttpAdapterInterface $httpAdapter)
```

Comment:
There is a glitch here, optional argument before a non optional argument.

Reply:
If we don't make it optional, would things break?